src/hotspot/share/memory/allocation.hpp

```diff
@@ -500,11 +500,11 @@ class AnyObj {
   void* operator new [](size_t size, const std::nothrow_t&  nothrow_constant, MEMFLAGS flags) throw() = delete;
 
   // Arena allocations
-  void* operator new(size_t size, Arena *arena) throw();
-  void* operator new [](size_t size, Arena *arena) throw() = delete;
+  void* operator new(size_t size, Arena *arena);
+  void* operator new [](size_t size, Arena *arena) = delete;
```

Comment:
`operator new[](size_t)` (down below, where github won't let me comment directly) should also have it's nothrow exception-spec removed.